lib/modules/manager/swift/artifacts.spec.ts

```diff
@@ -0,0 +1,202 @@
+import { join } from 'upath';
+import { envMock, mockExecAll } from '../../../../test/exec-util';
+import { env, fs, git } from '../../../../test/util';
+import { GlobalConfig } from '../../../config/global';
+import type { RepoGlobalConfig } from '../../../config/types';
+import * as docker from '../../../util/exec/docker';
+import type { UpdateArtifactsConfig } from '../types';
+import * as swift from '.';
+
+jest.mock('../../../util/exec/env');
+jest.mock('../../../util/git');
+jest.mock('../../../util/http');
+jest.mock('../../../util/fs');
+
+const config: UpdateArtifactsConfig = {};
+
+const adminConfig: RepoGlobalConfig = {
+  // `join` fixes Windows CI
+  localDir: join('/tmp/github/some/repo'),
+  cacheDir: join('/tmp/cache'),
+  containerbaseDir: join('/tmp/cache/containerbase'),
+};
+
+describe('modules/manager/swift/artifacts', () => {
+  beforeEach(() => {
+    jest.resetAllMocks();
+    jest.resetModules();
+
+    env.getChildProcessEnv.mockReturnValue(envMock.basic);
+    GlobalConfig.set(adminConfig);
+    docker.resetPrefetchedImages();
+  });
+
+  afterEach(() => {
+    GlobalConfig.reset();
+  });
+
+  it('returns null if no Package.resolved file found', async () => {
+    fs.getSiblingFileName.mockReturnValueOnce('Package.resolved');
+    fs.localPathExists.mockResolvedValueOnce(false);
+    const updatedDeps = [
+      {
+        depName: 'dep1',
+      },
+    ];
+    expect(
+      await swift.updateArtifacts({
+        packageFileName: 'Package.swift',
+        updatedDeps,
+        newPackageFileContent: '',
+        config,
+      })
+    ).toBeNull();
+  });
+
+  it('returns null if updatedDeps is empty', async () => {
+    expect(
+      await swift.updateArtifacts({
+        packageFileName: 'Package.swift',
+        updatedDeps: [],
+        newPackageFileContent: '',
+        config,
+      })
+    ).toBeNull();
+  });
+
+  it('returns null if unchanged', async () => {
+    fs.getSiblingFileName.mockReturnValueOnce('Package.resolved');
+    fs.localPathExists.mockResolvedValueOnce(true);
+    fs.readLocalFile.mockResolvedValueOnce('Current Package.resolved');
+    fs.getParentDir.mockReturnValueOnce('');
+    const execSnapshots = mockExecAll();
+    fs.readLocalFile.mockResolvedValueOnce('Current Package.resolved');
+
+    const updatedDeps = [
+      {
+        depName: 'dep1',
+      },
+    ];
+    expect(
+      await swift.updateArtifacts({
+        packageFileName: 'Package.swift',
+        updatedDeps,
+        newPackageFileContent: '',
+        config,
+      })
+    ).toBeNull();
+    expect(execSnapshots).toMatchSnapshot();
```

Comment:
use `toMatchObject` or `toEqual` instead of snapshots

Reply:
I attempted to address this in 7f398bab5df2dfb9e69296e570a4bd9874541159 - let me know if that matches what you had in mind.

Reply:
please just resolve conversations instead of comment to reduce noise.

we don't review single commits, we just check what's last changed on GitHub UI